src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,58 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exception.RegistrationException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user == null) {
+            throw new RegistrationException("The user doesn't exist");
+        }
+        validateLogin(user.getLogin());
+        validatePassword(user.getPassword());
+        validateAge(user.getAge());
```

Comment:
it is better to pass `user` instance to each method so that it is responsibility of a method to check what's necessary

caller should not bother about it
```suggestion
        validateLogin(user);
        validatePassword(user);
        validateAge(user);
```

Reply:
Could you explain why it is better please? 
When I pass user I should in the whole method use for example user.getLogin() while I could use just login instead